orchestrator-salt/src/main/java/com/sequenceiq/cloudbreak/orchestrator/salt/SaltOrchestrator.java

```diff
@@ -353,15 +356,34 @@ public void initServiceRun(List<GatewayConfig> allGateway, Set<Node> allNodes, S
             Callable<Boolean> saltPillarRunner = saltRunner.runner(hostSave, exitCriteria, exitModel);
             saltPillarRunner.call();
 
+            // Observation: The only pillar entries which seem to change when adding new nodes are disks and nodes.
+            //  PillarSave isn't very expensive - but the entire set of pillarSaves does take over 1 second.
             for (Entry<String, SaltPillarProperties> propertiesEntry : saltConfig.getServicePillarConfig().entrySet()) {
                 OrchestratorBootstrap pillarSave = new PillarSave(sc, gatewayTargetIpAddresses, propertiesEntry.getValue());
                 saltPillarRunner = saltRunner.runner(pillarSave, exitCriteria, exitModel);
                 saltPillarRunner.call();
             }
 
+            // Each grainAddRunner takes at least 0.6 seconds (combination of grains.append + grains.get). Occasionally longer than 1 second.
```

Comment:
The additional get is there to make sure the grains are added, especially on YCloud, because we observed many issues there Salt saying it's added, but actually it wasn't.

Reply:
OK. Not planning to change much here in the first patch.